core/block/editor/smartblock/smartblock.go

```diff
@@ -1346,6 +1351,40 @@ func hasDetailsMsgs(msgs []simple.EventMessage) bool {
 	return false
 }
 
+func (sb *smartBlock) saveActiveViews(msgs []simple.EventMessage) []simple.EventMessage {
+	i := 0
+	for _, msg := range msgs {
+		ev := msg.Msg.GetBlockDataviewActiveViewIsSet()
+		if ev != nil {
+			err := sb.objectStore.SetActiveView(sb.Id(), ev.BlockId, ev.ViewId)
+			if err != nil {
+				log.With("objectID", sb.Id()).Warnf("failed to update active view: %v", err)
+			}
+		} else {
+			msgs[i] = msg
+			i++
+		}
+	}
+	// we exclude BlockDataViewActiveViewIsSet messages, because we do not need to send events on it
+	return msgs[:i]
+}
+
+func (sb *smartBlock) injectActiveViews(s *state.State) {
+	if err := s.Iterate(func(b simple.Block) (isContinue bool) {
+		if d := b.Model().GetDataview(); d != nil {
```

Comment:
it feels like, instead of iterating all blocks for all smartblocks when opening them the reverse approach may be more efficient and also utilise the objectstore indexes.
- get all saved activeViews from objectstore for specific objectId
- then pick specific blocks and modify them

Reply:
Yes, I have rewritten this logic, so we store separate map-like string for each object